lib/pbench/server/api/resources/datasets_inventory.py

```diff
@@ -53,27 +53,35 @@ def _get(self, params: ApiParams, _) -> Response:
 
         GET /api/v1/datasets/inventory/{dataset}/{path}
         """
+        try:
+            path = params.uri["path"]
+        except KeyError:
+            path = None
```

Comment:
Instead, `path = params.uri.get("path")` (which defaults to `None` if the key doesn't exist).